data/blog/improving-developer-experience-from-unreadable-css-selectors-to-clear-component-names.md

```diff
@@ -0,0 +1,69 @@
+
+## Adding React component names into Sentry
+
+There were many opportunities across the product where we could leverage component names: spans, breadcrumbs, rage click issues, and [dead and rage click selectors](https://blog.sentry.io/introducing-rage-and-dead-click-detection-for-session-replay/).
+
+![React component name used in spans](/images/improving-developer-experience-from-unreadable-css-selectors-to-clear-component-names/span.png)
+
+_In spans, React component names are used in click interactions._
+
+![React component name used in replay breadcrumbs](/images/improving-developer-experience-from-unreadable-css-selectors-to-clear-component-names/replay-breadcrumb.png)
+
+_In Replay breadcrumbs, React component names are used in the selector path in user click breadcrumbs. If the user clicked on a component with a component name, you can search replays by that component name._
+
+![React component name used in rage and dead click selectors](/images/improving-developer-experience-from-unreadable-css-selectors-to-clear-component-names/rage-and-dead-selector.png)
+
+_In rage and dead click selectors, React component names are used as part of the selector name._
+
+![React component name used in rage click issues](/images/improving-developer-experience-from-unreadable-css-selectors-to-clear-component-names/rage-click-issue.png)
+
+_In rage click issues, the React component name is also shown if it’s available._
+
+The SDK had to be updated first to take in these new attributes. This involved creating a function to extract the component name from the attributes and propagating it throughout spans, UI breadcrumbs, and Replay breadcrumbs. Next, we updated Sentry’s UI to utilize these component names. In most areas, the component names just replaced the selector names, but for rage click issues, and dead and rage click selectors this was not the case. For rage click issues, the Clicked Element uses the selector name which provides useful context about the element. Hence decided to add an additional field for the component name: React Component Name. For rage and dead clicks, additional context besides the React component names are useful for debugging – for instance, `aria-label` and `data-test-id`. For that reason we decided to keep the selectors names. Since class names become minified, we decided to replace them with the react component name. Hence, rage and dead click selectors are now a combination of the component name and the additional context.
```

Comment:
```suggestion
The SDK had to be updated first to take in these new attributes. This involved creating a function to extract the component name from the attributes and propagating it throughout spans, UI breadcrumbs, and Replay breadcrumbs. Next, we updated Sentry’s UI to utilize these component names. In most areas, the component names just replaced the selector names, but for rage click issues, and dead and rage click selectors this was not the case. For rage click issues, the Clicked Element uses the selector name which provides useful context about the element. Hence we decided to add an additional field for the component name: React Component Name. For rage and dead clicks, additional context besides the React component names are useful for debugging – for instance, `aria-label` and `data-test-id`. For that reason we decided to keep the selectors names. Since class names become minified, we decided to replace them with the react component name and because of that, rage and dead click selectors are now a combination of the component name and the additional context.
```